### ISSUE TITLE
[Feature] Save optimizer-related params in best epoch

### ISSUE THREAD

**Issue by Tau-J** [open-mmlab/mmengine#1409]
### What is the feature?

I notice that mmengine does not save optimizer-related params, e.g. learning rate, so user cannot resume training from the best checkpoint. However, in an industrial development, it is an important feature to resume training from the last best checkpoint.

**Comment by joihn** [comment]
@HAOCHENYE 
I suffered from the same issue. I think it's very unexpected from this class and should be considered a bug.

This line :
https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/checkpoint_hook.py#L564

Should reuse the same `self.attribute` as this line 
https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/checkpoint_hook.py#L447

**Comment by joihn** [comment]
I did a PR:
https://github.com/open-mmlab/mmengine/pull/1454
